### DISCUSSION TITLE
How to run WinUI 3.0 unpackaged application elevated/as Administrator?

### DISCUSSION THREAD

**Discussion by optimozhu** [microsoft/WindowsAppSDK#3038] [Q&A]
I tried append properties like wpf, but it does not work.
`<requestedExecutionLevel level="requireAdministrator" uiAccess="false" />`
What should I do?

**Comment by DarranRowe** [comment]
The problem with the phrase "does not work" is that it can literally mean anything.
To you, it is obvious because you experienced it, but think of it from the perspective people answering the question. If someone was to say to you, "Could you help me, my car doesn't start. I tried hitting the green button, but it does not work." Would you be able to provide an answer?
The ability to answer the question is based upon the information that you give. In this case, there are three problems.
1) There is nothing about what environment you are working under. Is this a C# project or is it a C++ project. The answers can be very different.
2) What have you tried/what are you trying precisely. Just giving a random manifest snippet doesn't give any information on how you are trying to apply it.
3) What do you mean by does not work? Do you get manifest errors? Does the application run at standard privileges? Does the application crash?

On my end, with a C++ project:

![Screenshot 2022-10-11 222743](https://user-images.githubusercontent.com/52577874/195201862-5c356349-3a94-424a-8871-1ed455711ace.png)

the shield over the executable icon indicates that it requires UAC elevation, so requestedExecutionLevel is working. What's more, this project, which is WinUI 3, displays the UI properly. This means that it is working on my end. So please provide more information.

**Reply by optimozhu** [reply]
Thank you for your patience! Because I am amateur, sorry for that. Here is my test project.
[EvelationTestApp.zip](https://github.com/microsoft/WindowsAppSDK/files/9760643/EvelationTestApp.zip)

1. I create a C# winUI 3.0 project in Visual Studio 2022. 
2. I followed this [doc](https://learn.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app#unpackaged-create-a-new-project-for-an-unpackaged-c-or-c-winui-3-desktop-app) to modify the project, letting it to be a unpackaged application.
3. I add "WindowsAppSDKSelfContained" to my csproj file, to make it self-contained.
4. I add something below into app.mainfest file, atttampting make it run elevated. Following this [answer](https://stackoverflow.com/questions/22808953/how-ca-i-run-a-wpf-application-with-elevated-permissions).
`<trustInfo xmlns="urn:schemas-microsoft-com:asm.v2">`
`    <security>`
`        <requestedPrivileges xmlns="urn:schemas-microsoft-com:asm.v3">`
`            <requestedExecutionLevel level="requireAdministrator" uiAccess="false" />`
`        </requestedPrivileges>`
`    </security>`
`</trustInfo>`
5. I click green run button in Visual Studio. It launches directly without showing UAC dialog. And in the process I do not have elevated privilege. Also there is not shield over the executable icon.

**Reply by DarranRowe** [reply]
Hmm, well, making the application self-contained seems to cause issues. When done in a C++ application, linking fails due to mt.exe failing. If the application isn't self-contained then it works.
This means that this is either intentional, or the generation of the registration free WinRT entries add trust information by mistake. This is something that someone from the dev team would have to answer.

**Reply by optimozhu** [reply]
Thank you very much.

**Comment by AdamBraden** [comment]
@DrusTheAxe or @mikebattista - can you take a look?

**Reply by DrusTheAxe** [reply]
@mikebattista @MikeHillberg @Scottj1s this is more up your alley than mine

**Comment by farag2** [comment]
I have the same problem.
